### PR TITLE
fix: do not cache module resolution during launchpad dependency detection

### LINKED ISSUE #26685: CT onboarding hangs waiting for dependency installation
### Current behavior

When dependencies have been installed during CT onboarding, the Cypress App continues to wait showing a spinner with the text "Waiting for you to install the dependencies...". It does not exit this condition and requires the Cypress App to be restarted to continue. After restart, the Front-end framework is detected and onboarding continues successfully.

image

### Desired behavior

Onboarding should automatically and dynamically check dependency installation as it did up until Cypress [12.7.0](https://docs.cypress.io/guides/references/changelog#12-7-0).

After choosing a front-end framework during CT onboarding, for instance `Next.js`, the Cypress App displays a command, such as `npm install -D next react react-dom`, together with a spinner with the text "Waiting for you to install the dependencies...".

Now if the command to install packages is executed in a separate terminal window, the spinner and text disappear. The message displaying the command to execute disappears and is replaced with a text saying "You've successfully installed all required dependencies"

image

### Test code to reproduce

```bash
git init
npm init -y
npm install cypress -D -E
npx cypress open
```

Select Component Testing
Select Next.js, click Next step

In a separate terminal window copy and paste
`npm install -D next react react-dom`

The Cypress App waits showing a spinner with the text "Waiting for you to install the dependencies...". This does not change even when the dependencies have been installed.

### Cypress Version

`12.11.0` (and `12.12.0`)

### Node version

`18.16.0`

### Operating System

Ubuntu `22.04` (also reproducible on Windows `11`)

### Debug Logs

```shell
`cypress:scaffold-config:frameworks` is not detecting the dependencies
```

### Other

- This is possibly related to changes in Cypress [12.8.0](https://docs.cypress.io/guides/references/changelog#12-8-0) made for issue https://github.com/cypress-io/cypress/issues/25782.

### Workaround

Kill then restart the Cypress App after installing dev dependencies.

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/26685

### Additional details

We were caching the module resolution during the deps detection step, so even after installing a dep, we don't detect it - classic caching issue 🤦 

Only an issue if a user is missing dependencies, which shouldn't be too common.

### Steps to test

Repro in the original issue. I made a React/Vite project, removed `react-dom`, then proceeded to setup Cypress CT. react-dom was not detected. I did `yarn add react-dom` in the project, and now it is correctly detected.

### How has the user experience changed?

Correctly detects newly added dependencies during onboarding and shows user.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/scaffold-config/src/frameworks.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export async function isDependencyInstalled (dependency: Cypress.CypressComponen
     // for module resolution.
     const resolvePackagePath = require('resolve-package-path')
 
-    const packageFilePath = resolvePackagePath(dependency.package, projectPath)
+    const packageFilePath = resolvePackagePath(dependency.package, projectPath, false)
```
Comment (lmiller1990) [inline packages/scaffold-config/src/frameworks.ts:21]: This is the fix, the third arg is whether to cache or not - we don't want caching, or it won't find the dependency, even if the user adds it.

File: system-tests/projects/react-vite-ts-unconfigured/package.json [hunk 2]
```diff
@@ -8,8 +8,7 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "react": "^18.0.0",
-    "react-dom": "^18.0.0"
```
Comment (lmiller1990) [inline system-tests/projects/react-vite-ts-unconfigured/package.json:12]: Intentionally removed to reproduce issue in End to End test.

File: cli/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 12.13.0
+
+_Released 05/23/2023 (PENDING)_
+
+**Bugfixes:**
+
+- Fixed an issue where newly installed dependencies are not correct detected during Component Testing setup. Addresses [#26685](https://github.com/cypress-io/cypress/issues/26685).
```
Comment (mike-plummer) [inline cli/CHANGELOG.md]:
```suggestion
- Fixed an issue where newly-installed dependencies would not be detected during Component Testing setup. Addresses [#26685](https://github.com/cypress-io/cypress/issues/26685).
```

File: packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts [hunk 4]
```diff
@@ -102,6 +115,32 @@ describe('scaffolding component testing', {
...
+        await ctx.actions.file.writeFileInProject(
+          ctx.path.join('node_modules', 'react-dom', 'package.json'),
+          JSON.stringify({
+            'version': '17.0.0',
+          }),
+        )
+      })
+
+      // Wait for new dependency detection polling
+      cy.wait(3000)
```
Comment (mike-plummer) [inline packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts]: Is the explicit `wait` necessary here? Could we bump up the timeout on the following command to compensate?

**Review by mike-plummer — APPROVED** [review]
LGTM, just a couple minor suggestions - will approve assuming those are addressed one way or the other

File: cli/CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 12.13.0
```
Comment (mike-plummer) [inline cli/CHANGELOG.md]: Should this be a patch version bump?

File: packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts (hunk 4 above)
Comment (lmiller1990) [reply packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts]: I'll try removing it, it shouldn't be necessary.

File: cli/CHANGELOG.md (hunk 5 above)
Comment (lmiller1990) [reply cli/CHANGELOG.md]: Right, yes. I'll change this.

**Review by warrensplayer — APPROVED** [review]
LGTM

**Pull request merged**